docs/01_nodeos/03_plugins/producer_plugin/10_block-producing-explained.md

```diff
@@ -4,87 +4,61 @@ content_title: Block Production Explained
-        b0 -> b1 -> b2 -> b3 -> b4 -> b5
-    } //cluster_chain
-
-} //digraph
+```
+         +------+     +------+       +------+     +------+
+      -->| BP-A |---->| BP-A |------>| BP-B |---->| BP-B |
+         +------+     | Peer |       | Peer |     +------+
+                      +------+       +------+
 ```
 
-`BP-A` will send block at `p` and,
-
-`BP-B` needs block at time `t` or otherwise will drop it.
+`BP-A` will send block at `l` and, `BP-B` needs block at time `t` or otherwise will drop it.
 
 If `BP-A`is producing 12 blocks as follows `b(lock) at t(ime) 1`, `bt 1.5`, `bt 2`, `bt 2.5`, `bt 3`, `bt 3.5`, `bt 4`, `bt 4.5`, `bt 5`, `bt 5.5`, `bt 6`, `bt 6.5` then `BP-B` needs `bt 6.5` by time `6.5` so it has `.5` to produce `bt 7`.
 
 Please notice that the time of `bt 7` minus `.5` equals the time of `bt 6.5` therefore time `t` is the last block time of `BP-A` and when `BP-B` needs to start its first block.
 
-## Example 1
-`BP-A` has 50% e, m = 200ms, c = 200ms, n = 0ms, a = 250ms:
-`BP-A` sends at (t-250ms) <-> `BP-A-Peer` processes for 200ms and sends at (t - 50ms) <-> `BP-B-Peer` processes for 200ms and sends at (t + 150ms) <-> arrive at `BP-B` 150ms too late.
-
-## Example 2
-`BP-A` has 40% e and m = 200ms, c = 200ms, n = 0ms, a = 300ms:
-(t-300ms) <-> (+200ms) <-> (+200ms) <-> arrive at `BP-B` 100ms too late.
+A block is produced and sent when either it reaches `m` or `n` or `p`. 
 
-## Example 3
-`BP-A` has 30% e and m = 200ms, c = 150ms, n = 0ms, a = 350ms:
-(t-350ms) <-> (+150ms) <-> (+150ms) <-> arrive at `BP-B` with 50ms to spare.
+Starting in Leap 4.0, blocks are propagated after block header validation. This means instead of `BP-A Peer` & `BP-B Peer` taking `m` time to validate and forward a block it only takes a small number of milliseconds to verify the block header and then forward the block.
 
-## Example 4
-`BP-A` has 25% e and m = 200ms, c = 125ms, n = 0ms, a = 375ms:
-(t-375ms) <-> (+125ms) <-> (+125ms) <-> arrive at `BP-B` with 125ms to spare.
+Starting in Leap 5.0, blocks in a round are started immediately after the completion of the previous block. Before 5.0, blocks were always started on `w` intervals and a node would "sleep" between blocks if needed. In 5.0, the "sleeps" are all moved to the end of the block production round. 
 
-## Example 5
-`BP-A` has 10% e and m = 200ms, c = 50ms, n = 0ms, a = 450ms:
-(t-450ms) <-> (+50ms) <-> (+50ms) <-> arrive at `BP-B` with 350ms to spare.
+## Example 1: block arrives 110ms early (zero network latency between BP and immediate peer)
+* `BP-A` has e = 120, n = 0ms, h = 5ms, a = 10ms
+* `BP-A` sends b1 at `t1-10ms` => `BP-A-Peer` processes `h=5ms`, sends at `t-5ms` => `BP-B-Peer` processes `h=5ms`, sends at `t-0ms` => arrives at `BP-B` at `t`.
+* `BP-A` starts b2 at `t1-10ms`, sends b2 at `t2-20ms` => `BP-A-Peer` processes `h=5ms`, sends at `t2-15ms` => `BP-B-Peer` processes `h=5ms`, sends at `t2-10ms` => arrives at `BP-B` at `t2-10ms`.
```

Comment:
I thought it started `b2` as soon as `b1` was completed, and that it didn't wait for `t1-10ms`

Reply:
It does start `b2` as soon as `b1` is complete. `b1` is complete at `t1-10ms` which is when `b2` starts.

Reply:
Why would b1 be complete at `t1-10ms`? Couldn't it reach the `m` limit before that?

Reply:
All the existing examples assume non-full blocks, none of them reach `m` limit. I'll add an example where `m` is reached.